patterns/multi-chart-orchestration/README.md

```diff
@@ -0,0 +1,130 @@
+
+When distributing operators that install Custom Resource Definitions (CRDs) alongside applications that create custom resources from those CRDs, orchestration is generally required. This ensures that all components are installed in the correct order and that the necessary dependencies are present for the application to function correctly.
+
+## KOTS HelmChart Weight and Helm Upgrade Flags
+
+The weight property in the Replicated [HelmChart](https://docs.replicated.com/reference/custom-resource-helmchart-v2) custom resource allows you to control the order in which Helm charts are installed. KOTS installs Helm charts based on the weight value in ascending order, meaning that charts with lower weight values are installed first.
+
+You can set the weight property to any negative or positive integer, or 0. By default, if no weight is specified, the weight is set to 0.
+
+[Infra HelmChart](https://github.com/replicatedhq/platform-examples/blob/main/applications/mlflow/kots/infra-chart.yaml)
+```yaml
+apiVersion: kots.io/v1beta2
+kind: HelmChart
+metadata:
+  name: infra
+spec:
+  chart:
+    name: infra
+    chartVersion: 0.1.0
+  weight: -10
+  helmUpgradeFlags:
+    - --wait
+    - --timeout
+    - 600s
+```
+
+[Mlflow HelmChart](https://github.com/replicatedhq/platform-examples/blob/main/applications/mlflow/kots/mlflow-chart.yaml)
+```yaml
+apiVersion: kots.io/v1beta2
+kind: HelmChart
+metadata:
+  name: mlflow
+spec:
+  chart:
+    name: mlflow
+    chartVersion: 0.3.0
+  weight: 10
+  helmUpgradeFlags:
+    - --wait
+    - --timeout
+    - 600s
+```
+
+In the above two snippets, the `infra` chart that we're installing with KOTS includes the Cloudnative-PG operator for Postgres as well as the Minio Operator. Because the weight in the `infra` chart is set to `-10` which is lower than a value of `10` that Mlflow uses, it will be installed first.
+
+In this example, we're also using `helmUpgradeFlags` to specify additional flags to pass to the Helm upgrade command for charts. These flags are passed in addition to any flags KOTS passes by default. The values specified here take precedence if KOTS already passes the same flag. We're setting the Helm `--wait` flag to ensure that Helm waits until all Kubernetes resources are in a ready state before marking the release as successful. Additionally, we're setting a timeout for the wait flag to define the maximum amount of time to wait for the resources to become ready.
+
+To see more info on resource orchestration in KOTS as well as the reference for `kind: HelmChart`, visit the following two articles:
+
+https://docs.replicated.com/vendor/orchestrating-resource-deployment
+
+https://docs.replicated.com/reference/custom-resource-helmchart-v2
+
+## Validate existence of CRDs with Helm Hooks
```

Comment:
If you think it makes sense, could link to this section which explains a little more about the support for Helm hooks for charts deployed with KOTS https://docs.replicated.com/vendor/helm-native-about#hooks

Reply:
thanks!